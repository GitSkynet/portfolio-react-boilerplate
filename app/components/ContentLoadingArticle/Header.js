import React from 'react';
import PropTypes from 'prop-types';

import NewSpeechBubble from '../SpeechBubble';

function Header(props) {
  return (
    <header>
      <NewSpeechBubble
        arrowHeight={'4vh'}
        backgroundColor="#575756"
        isLeftHanded={false}
        showArrowBreakpoint="750px"
      >
        {props.children}
      </NewSpeechBubble>
    </header>
  );
}

Header.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
};

export default Header;
