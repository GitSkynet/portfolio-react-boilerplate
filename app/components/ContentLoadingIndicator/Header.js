import React from 'react';
import PropTypes from 'prop-types';

import NewSpeechBubble from '../SpeechBubble';
import { theme } from '../../styles/theme';

import { BASE_LINE_HEIGHT } from '../../styles/typography';

const { backgroundColor } = theme.loader;

function Header(props) {
  return (
    <header>
      <NewSpeechBubble
        arrowHeight={'4vh'}
        backgroundColor={backgroundColor}
        isLeftHanded={false}
        padding={`${BASE_LINE_HEIGHT}rem`}
        showArrowBreakpoint="750px"
      >
        {props.children}
      </NewSpeechBubble>
    </header>
  );
}

Header.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
};

export default Header;
