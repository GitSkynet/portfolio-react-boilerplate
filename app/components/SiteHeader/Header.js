import styled from 'styled-components';

const Header = styled.header`
  flex: none;
  z-index: 10;
  background-color: #465a65;
`;

export default Header;
