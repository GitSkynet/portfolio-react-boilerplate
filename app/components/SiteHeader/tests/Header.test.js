import React from 'react';
import { shallow } from 'enzyme';

import Header from '../Header';

describe('<Header />', () => {
  it('should render and match the snapshot', () => {
    const renderedComponent = shallow(<Header />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should adopt the "backgroundColor" property', () => {
    const color = '#bada55';
    const renderedComponent = shallow(<Header backgroundColor={color} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
