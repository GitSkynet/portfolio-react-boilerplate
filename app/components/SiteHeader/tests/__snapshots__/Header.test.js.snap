// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render and match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header__Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "Header__Header-gbjpXo kBDQvX",
    },
    "ref": null,
    "rendered": null,
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "Header__Header-gbjpXo kBDQvX",
      },
      "ref": null,
      "rendered": null,
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
